bridgerelayer/substrate/client.go

```diff
@@ -0,0 +1,131 @@
+package substrate
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	gsrpc "github.com/Snowfork/go-substrate-rpc-client"
+	"github.com/Snowfork/go-substrate-rpc-client/config"
+	"github.com/Snowfork/go-substrate-rpc-client/signature"
+	"github.com/Snowfork/go-substrate-rpc-client/types"
+
+	etypes "github.com/snowfork/polkadot-ethereum/bridgerelayer/types"
+)
+
+// Client struct
+type Client struct {
+	api      *gsrpc.SubstrateAPI
+	metadata *types.Metadata
+}
+
+// NewClient foo
+func NewClient() (*Client, error) {
+
+	api, err := gsrpc.NewSubstrateAPI(config.Default().RPCURL)
+	if err != nil {
+		return nil, err
```

Comment:
Here we return the error, but on line 31 we panic. Returning an error allows us to gracefully handle an expected situation while panic shuts down the program. Generally, panic should be used for initialization/set up and any error which results in an unrecoverable state. Both error and panic are potentially valid depending on where initialization of the Substrate client takes place, but in any given context the error handling strategy should be consistent e.g. both panic or both return error.

Reply:
Yeh, the client code was a bit half-baked in that regard. I've fixed it in latest upload and there is no more panicking.